### PR TITLE
pj-rehearse: choose periodics and presubmits with changed clusters

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @petr-muller @hongkailiu

### CODE REVIEW

File: pkg/diffs/diffs.go [hunk 1]
```diff
@@ -147,6 +147,32 @@ func getJobsByRepoAndName(presubmits config.Presubmits) map[string]map[string]pr
 	return jobsByRepo
 }
 
+// GetChangedClusterJobs returns presubmits and periodics that have had their cluster field changed.
+func GetChangedClusterJobs(prowMasterConfig, prowPRConfig *prowconfig.Config, logger *logrus.Entry) (config.Presubmits, config.Periodics) {
+	presubmits := config.Presubmits{}
+	periodics := config.Periodics{}
+
+	masterJobs := getJobsByRepoAndName(prowMasterConfig.JobConfig.PresubmitsStatic)
```
Comment (hongkailiu) [inline pkg/diffs/diffs.go]: NIP: `masterJobs` -> `masterPresubmits`?

File: pkg/diffs/diffs_test.go [hunk 2]
```diff
@@ -916,3 +916,69 @@ func makeConfigWithPeriodics(p []prowconfig.Periodic) *prowconfig.Config {
...
+				if err := deepcopy.Copy(&b, baseConfig); err != nil {
+					t.Fatal(err)
+				}
+				return &a, &b
+			},
+			expectedPresubmits: config.Presubmits{},
+			expectedPeriodics:  config.Periodics{},
+		},
+		{
+			name: "all cluster changes, nothing to rehearse",
```
Comment (hongkailiu) [inline pkg/diffs/diffs_test.go]: `nothing` -> `all`?

**Comment by stevekuznetsov** [comment]
@hongkailiu updated

File: pkg/diffs/diffs.go [hunk 3]
```diff
@@ -147,6 +147,32 @@ func getJobsByRepoAndName(presubmits config.Presubmits) map[string]map[string]pr
 	return jobsByRepo
 }
 
+// GetChangedClusterJobs returns presubmits and periodics that have had their cluster field changed.
+func GetChangedClusterJobs(prowMasterConfig, prowPRConfig *prowconfig.Config, logger *logrus.Entry) (config.Presubmits, config.Periodics) {
```
Comment (petr-muller) [inline pkg/diffs/diffs.go]: @stevekuznetsov I realize I'm coming late, but is there a reason why the cluster comparison couldn't be added to the `GetChanged{Presubmits,Periodics}`?

**Pull request merged**